src/hive_mind_bridge/test/SocketToMoveByIntegrationTest.cpp

```diff
@@ -116,26 +116,24 @@ int main(int argc, char** argv) {
              "\tPayload: %d",
              statusReturnFunctionName.c_str(), arg0);
 
-    // Send moveBy request
-    serializer.serializeToStream(moveByMessageDTO);
-    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
-
-    // Listen for a response
-    MessageDTO message;
-    deserializer.deserializeFromStream(message);
-    ResponseDTO response = std::get<ResponseDTO>(message.getMessage());
-    UserCallResponseDTO userCallResponse = std::get<UserCallResponseDTO>(response.getResponse());
-    FunctionCallResponseDTO functionCallResponse =
-        std::get<FunctionCallResponseDTO>(userCallResponse.getResponse());
-    GenericResponseDTO genericResponse = functionCallResponse.getResponse();
-    GenericResponseStatusDTO status = genericResponse.getStatus();
-    std::string details = genericResponse.getDetails();
-    ROS_INFO("RESPONSE FROM HOST (moveBy): \n"
-             "\tResponse status: %d\n"
-             "\tDetails: %s",
-             status, details.c_str());
-
-    // Listen for a return message
-    MessageDTO returnMessage;
-    deserializer.deserializeFromStream(returnMessage);
+    for (int i = 0; i < 5; i++) {
+        // Send moveBy request
+        serializer.serializeToStream(moveByMessageDTO);
+
+        // Listen for a response
+        MessageDTO message;
+        deserializer.deserializeFromStream(message);
+        ResponseDTO response = std::get<ResponseDTO>(message.getMessage());
```

Comment:
Maybe verify what kind of msg the deserializer created. If it is not a response, there will probably be a nullptr error

Reply:
Yes and that's precisely the point. This is a test main. I want it to throw if there is a problem.

Reply:
my bad, I didn't see it was in the test folder